homeassistant/components/netatmo/__init__.py

```diff
@@ -151,6 +154,21 @@ def _service_dropwebhook(service):
         schema=SCHEMA_SERVICE_DROPWEBHOOK,
     )
 
+    home_data = pyatmo.HomeData(auth)
```

Comment:
`pyatmo.HomeData` init will raise Exceptions if no thermostat (only cameras or weather station for instance), see [here](https://github.com/jabesq/netatmo-api-python/blob/7c91d7cb859d5d2dace691760ea561fabbfa92d0/src/pyatmo/thermostat.py#L16-L35).

You should add this in try catch after the `auth` init (line 111), or, in the service.

Reply:
Fixed - added it to try. The constructor of `HomeData` looks complex, semed like a waste to recreate it each time the service gets called.